transcom/mymove
# B-17918: remove old PPM INTERNAL APIs - SET A

## [B-17918](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A868169)

[E-04480](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A833435)

## Summary
This PR is for the deprecation of the following INTERNAL APIs:
`/moves/{moveId}/personally_procured_move - GET`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/indexPersonallyProcuredMoves

`/moves/{moveId}/personally_procured_move - POST`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/createPersonallyProcuredMove

`/moves/{moveId}/personally_procured_move/{personallyProcuredMoveId} - GET`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/showPersonallyProcuredMove

`/moves/{moveId}/personally_procured_move/{personallyProcuredMoveId} - PUT`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updatePersonallyProcuredMove

NOTE: All references to the deprecated APIs were removed both on the GO and client-side. To mitigate risk, all targeted APIs marked for deprecation were grouped into two branches for [B-17918](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A868169): 

- B-17918-Remove-create_ppm_attachments-from-Internal-API-A

- B-17918-Remove-create_ppm_attachments-from-Internal-API-B ([PR](https://github.com/transcom/mymove/pull/11678))

**This PR is for branch: B-17918-Remove-create_ppm_attachments-from-Internal-API-A**

The [deprecation process](https://transcom.github.io/mymove-docs/docs/api/guides/how-to-deprecate-endpoints/) was not observed because it's INTERNAL and not Prime ([per Dre](https://caci-bits.slack.com/archives/C05DX3AG6EL/p1701194838343559)).

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
The following have been removed from Swagger:

`/moves/{moveId}/personally_procured_move - GET`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/indexPersonallyProcuredMoves

`/moves/{moveId}/personally_procured_move - POST`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/createPersonallyProcuredMove

`/moves/{moveId}/personally_procured_move/{personallyProcuredMoveId} - GET`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/showPersonallyProcuredMove

`/moves/{moveId}/personally_procured_move/{personallyProcuredMoveId} - PUT`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updatePersonallyProcuredMove

1. Access Internal Swagger: http://milmovelocal:3000/swagger-ui/internal.htm
2. http://milmovelocal:3000/swagger-ui/internal.html#/ppm/indexPersonallyProcuredMoves (does not exist)
3. http://milmovelocal:3000/swagger-ui/internal.html#/ppm/createPersonallyProcuredMove  (does not exist)
4. http://milmovelocal:3000/swagger-ui/internal.html#/ppm/showPersonallyProcuredMove  (does not exist)
5. http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updatePersonallyProcuredMove  (does not exist)
6. Log into http://milmovelocal:3000/  (not see any React build errors, or runtime errors navigating the application)
### Frontend



## Screenshots
The following have been removed from Swagger:

`/moves/{moveId}/personally_procured_move - GET`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/indexPersonallyProcuredMoves

`/moves/{moveId}/personally_procured_move - POST`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/createPersonallyProcuredMove

`/moves/{moveId}/personally_procured_move/{personallyProcuredMoveId} - GET`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/showPersonallyProcuredMove

`/moves/{moveId}/personally_procured_move/{personallyProcuredMoveId} - PUT`

- http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updatePersonallyProcuredMove
